fossunited/payments/doctype/razorpay_payment/test_razorpay_payment.py

```diff
@@ -0,0 +1,136 @@
+                "event_end_date": "2024-09-05 13:00:00",
+                "event_description": fake.paragraph(),
+                "t_shirt_price": 100,  # Set t-shirt price
+                "is_paid_event": 1,
+                "tiers": [{"title": "General", "price": 200}],
+            }
+        ).insert(ignore_permissions=True)
+        return event.name
+
+    def create_payment_document(self, user):
+        frappe.set_user(user.name)
+        tier = frappe.get_doc(EVENT, self.event).tiers[0]
+
+        # Have taken this metadata to mimic the razorpay payment metadata
+        meta_data = {
+            "attendees": [
+                {
+                    "designation": fake.job(),
+                    "email": fake.email(),
+                    "full_name": fake.name(),
+                    "organization": fake.company(),
+                    "placeholder": fake.name(),
+                    "wants_tshirt": 0,
+                },
+                {
+                    "designation": fake.job(),
+                    "email": fake.email(),
+                    "full_name": fake.name(),
+                    "organization": fake.company(),
+                    "placeholder": fake.name(),
+                    "wants_tshirt": 0,
+                },
+            ],
+            "custom_fields": {
+                "food_preference": fake.random_element(elements=("Veg", "Non-Veg", "Vegan"))
+            },
+            "event": self.event,
+            "num_seats": "2",
+            "tier": {
+                "creation": str(tier.creation),
+                "currency": tier.currency,
+                "description": tier.description,
+                "docstatus": tier.docstatus,
+                "doctype": "FOSS Ticket Tier",
+                "enabled": tier.enabled,
+                "idx": tier.idx,
+                "maximum_tickets": tier.maximum_tickets,
+                "modified": str(tier.modified),
+                "modified_by": tier.modified_by,
+                "name": tier.name,
+                "owner": tier.owner,
+                "parent": tier.parent,
+                "parentfield": "tiers",
+                "parenttype": EVENT,
+                "price": tier.price,
+                "title": tier.title,
+                "valid_till": str(tier.valid_till),
+            },
+        }
+
+        invoice = frappe.get_doc(
+            {
+                "doctype": RAZORPAY_PAYMENT,
+                "document_type": EVENT,
+                "document_name": self.event,
+                "email": user.email,
+                "amount": tier.price * 2,  # 2 attendees
+                "currency": "INR",
+                "status": "Pending",
+                "meta_data": json.dumps(meta_data),
+            }
+        )
+
+        try:
+            invoice.insert()
+            frappe.db.commit()
+            return invoice
+        except frappe.PermissionError:
+            frappe.db.rollback()
+            return None
```

Comment:
CI is failing because of this block. You need to throw the error rather than returning `None`